#goal definition
string[] databasepaths
string goal_path
---
#result definition
int result
---
#feedback
int state